conanfile.py

```diff
@@ -75,8 +85,46 @@ class Xrpl(ConanFile):
         'zlib/1.2.11',
     ]
 
+    def set_version(self):
+        BUILDINFO = f"src/ripple/protocol/impl/BuildInfo.cpp"
+        regex = r"versionString\s?=\s?\"(.*)\""
+        with open(BUILDINFO, 'r') as file:
+            while True:
+                if match := re.search(regex, file.readline()): break
+            self.version =  match.group(1)
```

Comment:
This (or my suggested previous line) will throw if a matching line is not found, but I guess that's ok.